### PR TITLE
remove Get-PSHostProcessInfo under non-windows platform

### LINKED ISSUE #5081: Get-PSHostProcessInfo fails on non-Windows
We may have to disable this cmdlet on non-Windows

Steps to reproduce
------------------

```powershell
Get-PSHostProcessInfo                                                                                             
```

Expected behavior
-----------------

```none
List PowerShell processes
```

Actual behavior
---------------

```none
Get-PSHostProcessInfo : Unable to load DLL 'api-ms-win-core-file-l1-1-0.dll': The specified module or one of its dependencies could not be found.
 (Exception from HRESULT: 0x8007007E)
At line:1 char:1
+ Get-PSHostProcessInfo
+ ~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-PSHostProcessInfo], DllNotFoundException
    + FullyQualifiedErrorId : System.DllNotFoundException,Microsoft.PowerShell.Commands.GetPSHostProcessInfoCommand

```

Environment data
----------------

```powershell
Name                           Value                                                                                                               
----                           -----                                                                                                               
PSVersion                      6.0.0-beta.8                                                                                                        
PSEdition                      Core                                                                                                                
GitCommitId                    v6.0.0-beta.8                                                                                                       
OS                             Darwin 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64      
Platform                       Unix                                                                                                                
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                             
PSRemotingProtocolVersion      2.3                                                                                                                 
SerializationVersion           1.1.0.1                                                                                                             
WSManStackVersion              3.0  
```

### PR DESCRIPTION
**Pull request by chunqingchen**
this is to resolve #5081

since api-ms-win-core-file-l1-1-0.dll is not supported under non-windows platform, remove the cmdlet

### CODE REVIEW

File: test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1 [hunk 1]
```diff
@@ -280,3 +280,17 @@ Describe "Remoting loopback tests" -Tags @('CI', 'RequireAdminOnWindows') {
         $result | Should Be 100
     }
 }
+
+Describe "Verify remoting cmdlets that should not be implemented under non-windows platform" -Tags @("CI") {
+    It "Get-PSHostProcessInfo should be removed under non-windows platform" -Skip $IsWindows {
```
Comment (iSazonov) [inline test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1]: We already have the test in [DefaultCommands.Tests.ps1](https://github.com/PowerShell/PowerShell/blob/master/test/powershell/engine/Basic/DefaultCommands.Tests.ps1#L277)

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 2]
```diff
@@ -5535,6 +5535,7 @@ internal static class PSSnapInHelpers
 #if !UNIX
                 {"Disable-PSRemoting",                new SessionStateCmdletEntry("Disable-PSRemoting", typeof(DisablePSRemotingCommand), helpFile) },
                 {"Enable-PSRemoting",                 new SessionStateCmdletEntry("Enable-PSRemoting", typeof(EnablePSRemotingCommand), helpFile) },
+                {"Get-PSHostProcessInfo",             new SessionStateCmdletEntry("Get-PSHostProcessInfo", typeof(GetPSHostProcessInfoCommand), helpFile) },
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: What about `Enter-PSHostProcess` and `Exit-PSHostProcess`? They are supposed to work with `Get-PSHostProcessInfo`. Maybe we should exclude them all? @PaulHigin thoughts?

File: test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1 (hunk 1 above)
Comment (daxian-dbw) [reply test/powershell/engine/Remoting/RemoteSession.Basic.Tests.ps1]: As @iSazonov pointed out, the test `DefaultCommands.Tests.ps1` covers available cmdlets on different plats by default, can you please update that test?

**Review by daxian-dbw — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/engine/InitialSessionState.cs (hunk 2 above)
Comment (PaulHigin) [reply src/System.Management.Automation/engine/InitialSessionState.cs]: Yes, these cmdlets should be exluded as well.

**Review by iSazonov — APPROVED** [review]

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**